### PR TITLE
basic functionality working

### PR DESCRIPTION
**Pull request by ingawei**
image

### CODE REVIEW

File: web/components/contract/like-item-button.tsx [hunk 1]
```diff
@@ -37,16 +38,20 @@ export function LikeItemButton(props: {
...
+        .then(() => setTempTip((tempTip) => tempTip - LIKE_TIP_AMOUNT))
+        .catch(() => setIsLiking(false))
     }, 3000)
     toast.custom(
       () => (
         <TipToast
           userName={item.creatorUsername}
           onUndoClick={() => {
             clearTimeout(timeoutId)
+            setTempTip((tempTip) => tempTip - LIKE_TIP_AMOUNT)
```
Comment (jahooma) [inline web/components/contract/like-item-button.tsx:54]: Unrelated changes?

File: web/components/widgets/pagination.tsx [hunk 2]
```diff
@@ -35,56 +36,124 @@ export function PaginationNextPrev(props: {
...
-  } = props
+  const { page, itemsPerPage, totalItems, setPage, scrollToTop, className } =
+    props
 
   const maxPage = Math.ceil(totalItems / itemsPerPage) - 1
 
   if (maxPage <= 0) return <Spacer h={4} />
 
+  const pageNumbers = getPageNumbers(maxPage, page)
+  console.log('page', page)
```
Comment (jahooma) [inline web/components/widgets/pagination.tsx:56]: Can remove

File: web/components/widgets/pagination.tsx [hunk 3]
```diff
@@ -35,56 +36,124 @@ export function PaginationNextPrev(props: {
...
-        onClickNext={() => setPage(page + 1)}
+      <PaginationArrow
+        scrollToTop={scrollToTop}
+        onClick={() => setPage(page - 1)}
+        disabled={page <= 0}
+        nextOrPrev="prev"
+      />
+      <Row className="gap-2">
+        {pageNumbers.map((pageNumber) => (
+          <PageNumbers pageNumber={pageNumber} setPage={setPage} page={page} />
```
Comment (jahooma) [inline web/components/widgets/pagination.tsx:73]: Should add a key prop: `key={pageNumber}`

File: web/components/widgets/pagination.tsx [hunk 4]
```diff
@@ -35,56 +36,124 @@ export function PaginationNextPrev(props: {
...
+}
+
+type pageNumbers = number | '...'
+
+export function getPageNumbers(
+  maxPage: number,
+  page: number
+): Array<pageNumbers> {
+  if (maxPage <= 7) {
+    return Array.from({ length: maxPage + 1 }, (_, index) => index)
```
Comment (jahooma) [inline web/components/widgets/pagination.tsx:145]: This works. You could also use `range(0, maxPage + 1)` from lodash

Ditto below.

File: web/components/widgets/tipper.tsx [hunk 5]
```diff
@@ -89,7 +89,7 @@ export function Tipper(prop: {
     <Row className="items-center gap-0.5">
       <TipButton
         tipAmount={LIKE_TIP_AMOUNT}
-        totalTipped={totalTip}
+        totalTipped={totalTip + tempTip}
```
Comment (jahooma) [inline web/components/widgets/tipper.tsx:92]: This file also has unrelated changes

File: web/components/comments/comments-list.tsx [hunk 6]
```diff
@@ -11,7 +11,8 @@ import { Col } from '../layout/col'
 import { Content } from '../widgets/editor'
 import { LoadingIndicator } from '../widgets/loading-indicator'
 import { UserLink } from 'web/components/widgets/user-link'
-import { PaginationNextPrev } from 'web/components/widgets/pagination'
+import { Pagination, PaginationNextPrev } from '../widgets/pagination'
+// import { PaginationNextPrev } from 'web/components/widgets/pagination'
```
Comment (jahooma) [inline web/components/comments/comments-list.tsx:15]: Can remove

**Review by jahooma — APPROVED** [review]
Thanks Inga! This is an immediate improvement today for our paginated sections, and probably we'll always have some need for this component.

File: web/components/widgets/pagination.tsx (hunk 4 above)
Comment (ingawei) [reply web/components/widgets/pagination.tsx]: I think there's some type issues with that cause it could be strings or numbers
